src/coreclr/src/System.Private.CoreLib/src/System/Threading/Interlocked.CoreCLR.cs

```diff
@@ -232,6 +232,15 @@ public static partial class Interlocked
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern void MemoryBarrier();
 
+        /// <summary>
+        /// Synchronizes memory access as follows:
+        /// The processor that executes the current thread cannot reorder instructions in such a way that memory reads before
+        /// the call to <see cref="LoadBarrier"/> execute after memory accesses that follow the call to <see cref="LoadBarrier"/>.
+        /// </summary>
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern void LoadBarrier();
```

Comment:
Should we turn this into a public API?

Reply:
Perhaps. Maybe add StoreBarrier for symmetry as well. I did not have a need for store fences in this change, so I did not add it.

Reply:
Nit: Other similar managed APIs use Read/Write - should this follow the convention? ie ReadMemoryBarrier / WriteMemoryBarrier ?

Reply:
`ReadMemoryBarrier` sounds good. 

These things are often called differently - Load, Read, Fence, Barrier. I was not sure what would be more consistent with the rest of APIs.

Reply:
Adding a public API for this seems reasonable, and I agree it'd be good to have symmetry.  Might be worth checking other places we use multiple volatile reads to see if it could be used there for similar gains on arm. `Read/WriteMemoryBarrier` sounds fine to me.